clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -1100,6 +1101,8 @@ void EmitAssemblyHelper::RunOptimizationPipeline(
       // Rerun aspect propagation without warning diagnostics.
       MPM.addPass(SYCLPropagateAspectsUsagePass(/*ExcludeAspects=*/{},
                                                 /*ValidateAspects=*/false));
+      // Remove SYCL metadata added by sycl-post-link tool, like sycl_aspects
```

Comment:
Metadata is added by front-end, if I am not mistaken.

Reply:
Yeah, my mistake, other metadata like sycl_used/declared_aspects are introduced by sycl-post-link